### PR TITLE
Add module for PBS Pro client hosts 

### PR DESCRIPTION
**Pull request by tpdownes**
### Submission Checklist

* [X] Have you installed and run this change against pre-commit? (`pre-commit
  install`)
* [X] Are all tests passing? (`make tests`)
* [X] Have you written unit tests to cover this change?
* [ ] Is unit test coverage still above 80%?
* [X] Have you updated all applicable documentation?
* [X] Have you followed the guidelines in our Contributing document?

### CODE REVIEW

File: community/modules/scheduler/pbspro-client/README.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
+## Description
+
+This module provisions 1 or more PBS Client hosts to submit jobs to a PBS
+Professional cluster. The following extra services are required:
+
+- an existing licensed PBS Professional Server; if provisioned in the cloud, we
+  recommend using the [pbspro-server module][pbspro-server].
+- a shared filesystem mounted across all PBS hosts to faciliate file transfers
+  for jobs and their stdin/stdout logs
```
Comment (heyealex) [inline community/modules/scheduler/pbspro-client/README.md:9]: nit: for consistency, there should be a period at the end of this line like there is in the line above.

File: community/modules/scheduler/pbspro-client/variables.tf [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "project_id" {
+  description = "Project in which Google Cloud Storage bucket will be created"
```
Comment (heyealex) [inline community/modules/scheduler/pbspro-client/variables.tf:18]: I think this needs to be updated, it's more than a bucket in this case.

File: community/modules/scheduler/pbspro-client/variables.tf [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+    scopes = set(string)
+  })
+  default = {
+    email = null
+    scopes = ["https://www.googleapis.com/auth/devstorage.read_write",
+      "https://www.googleapis.com/auth/logging.write",
+      "https://www.googleapis.com/auth/monitoring.write",
+      "https://www.googleapis.com/auth/servicecontrol",
+      "https://www.googleapis.com/auth/service.management.readonly",
+    "https://www.googleapis.com/auth/trace.append"]
```
Comment (heyealex) [inline community/modules/scheduler/pbspro-client/variables.tf:157]: Odd formatting by tflint here. No need to fix it, but I'm wondering if putting the square brackets on their own lines would fix this?

Feel free to disregard.

File: community/modules/scheduler/pbspro-client/README.md (hunk 1 above)
Comment (tpdownes) [reply community/modules/scheduler/pbspro-client/README.md]: Addressed on HEAD of PR

File: community/modules/scheduler/pbspro-client/variables.tf (hunk 2 above)
Comment (tpdownes) [reply community/modules/scheduler/pbspro-client/variables.tf]: Addressed on HEAD of PR

File: community/modules/scheduler/pbspro-client/variables.tf (hunk 3 above)
Comment (tpdownes) [reply community/modules/scheduler/pbspro-client/variables.tf]: This format made its way into a couple places in other areas. Addressed on HEAD of PR

**Review by heyealex — APPROVED** [review]

**Pull request merged**